app/actions/remote/custom_emoji.test.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import DatabaseManager from '@database/manager';
+import NetworkManager from '@managers/network_manager';
+import {getFullErrorMessage} from '@utils/errors';
+import {logDebug} from '@utils/log';
+
+import {fetchCustomEmojis, searchCustomEmojis, fetchCustomEmojiInBatchForTest} from './custom_emoji';
+
+jest.mock('@managers/network_manager');
+jest.mock('@utils/log');
+jest.mock('@utils/errors');
+
+const serverUrl = 'baseHandler.test.com';
+
+const emojiId = 'emoji_id';
+const emoji = {id: emojiId, name: 'emoji_name'};
+const mockEmojis = [emoji];
+const error = new Error('Test error');
+
+beforeEach(async () => {
+    jest.clearAllMocks();
+    await DatabaseManager.init([serverUrl]);
+});
+
+describe('fetchCustomEmojis', () => {
+    it('should fetch custom emojis successfully', async () => {
+        const mockClient = {
+            getCustomEmojis: jest.fn().mockResolvedValue(mockEmojis),
+        };
+        (NetworkManager.getClient as jest.Mock).mockReturnValue(mockClient);
+
+        const result = await fetchCustomEmojis(serverUrl);
+
+        expect(result).toEqual({data: mockEmojis});
+        expect(NetworkManager.getClient).toHaveBeenCalledWith(serverUrl);
+        expect(mockClient.getCustomEmojis).toHaveBeenCalled();
+    });
+
+    it('should handle error during fetch custom emojis', async () => {
+        const mockClient = {
+            getCustomEmojis: jest.fn().mockRejectedValue(error),
+        };
+        (NetworkManager.getClient as jest.Mock).mockReturnValue(mockClient);
+        (getFullErrorMessage as jest.Mock).mockReturnValue('Full error message');
+
+        const result = await fetchCustomEmojis(serverUrl);
+
+        expect(result).toEqual({error});
+        expect(NetworkManager.getClient).toHaveBeenCalledWith(serverUrl);
+        expect(mockClient.getCustomEmojis).toHaveBeenCalled();
+        expect(logDebug).toHaveBeenCalledWith('error on fetchCustomEmojis', 'Full error message');
+    });
```

Comment:
would it help to also check if _forceLogoutIfNecessary).toHaveBeenCalledWith_
forceLogoutIfNecessary is covered in sessios tests.

Reply:
Good idea, added.